pulsar-client-cpp/lib/ClientConnection.cc

```diff
@@ -114,13 +114,13 @@ isTlsAllowInsecureConnection_(false) {
     if (clientConfiguration.isUseTls()) {
         using namespace boost::filesystem;
 
-        boost::asio::ssl::context ctx(boost::asio::ssl::context::tlsv12_client);
+        boost::asio::ssl::context ctx(executor_->io_service_, boost::asio::ssl::context::tlsv1_client);
 
```

Comment:
I mean, at least, we could add an `#ifdef` block to select the TLS version depending on the boost version.